agent/consul/connect_ca_endpoint.go

```diff
@@ -226,6 +226,18 @@ func (s *ConnectCA) ConfigurationSet(
 	// either by swapping the provider type or changing the provider's config
 	// to use a different root certificate.
 
+	// First up, sanity check that the current provider actually supports
+	// cross-signing.
+	oldProvider, _ := s.srv.getCAProvider()
+	if oldProvider == nil {
+		return fmt.Errorf("internal error: CA provider is nil")
+	}
+	if !oldProvider.SupportsCrossSigning() && !args.Config.ForceWithoutCrossSigning {
+		return errors.New("The current CA Provider does not support cross-signing. " +
+			"You can try again with ForceWithoutCrossSigningSet but this may cause " +
+			"disruption - see documentation for more.")
```

Comment:
Would it make sense to drop an INFO or WARN message to the logger when the provider does not support cross-signing and the force flag is set? That way there would be a very small log trail of when the override was explicitly used.

Reply:
Good call.